src/Monotonic.cpp

```diff
@@ -443,7 +443,12 @@ class DerivativeBounds : public IRVisitor {
             } else {
                 delta.accept(this);
                 ConstantInterval rdelta = result;
-                adjusted_delta = multiply(rcond, rdelta);
+                // If rdelta is constant, we can't trust it, because if it were truly
+                // constant, find_constant_bounds of the delta would have produced a
+                // bounded interval.
+                if (!is_constant(rdelta)) {
```

Comment:
The obvious thing to check is is_constant(rcond), why not that?

Reply:
Are you saying this is overly conservative in the case that rcond is constant? That this if should be `is_constant(rcond) || !is_constant(rdelta)` ?

Reply:
No, it's that I don't understand why !is_constant(rdelta) is correct. I can see that there's a failing case where it's constant, but I don't see why this captures all the failing cases.

Reply:
Can you propose something specific? Otherwise I am not sure how to go forward here.

I can't think of any cases that fail with this. While this seems like maybe a hack, I think it might actually be correct. In order for something to slip through here, it would need to be unbounded but also not constant, and still produce an incorrect rdelta somehow.